### PR TITLE
Update the completed profiles validator schema

### LINKED ISSUE #356: Update the completed profiles validation schema
- [ ] update the validation schema to check on the required data for the intern to be able to make a booking request `server/middlewares/validation/internCompleteProfileSchema.js`

### PR DESCRIPTION
**Pull request by RamyAlshurafa**
Fixes #356

### CODE REVIEW

File: server/middlewares/validation/internCompleteProfileSchema.js [hunk 1]
```diff
@@ -1,37 +1,72 @@
...
+  internshipOfficeAddress: Joi.string().required(),
+  emergencyContact: Joi.object({
     name: Joi.string().required(),
-    email: Joi.string().required(),
-  }).required(),
-  reference2: Joi.object({
-    name: Joi.string().required(),
-    email: Joi.string().required(),
-  }).required(),
+    email: Joi.required(),
```
Comment (ashatat) [inline server/middlewares/validation/internCompleteProfileSchema.js]: Is it ok to add a non-valid email here?

```suggestion
    email: Joi.string()
      .email()
      .required(),
```

File: server/middlewares/validation/internCompleteProfileSchema.js [hunk 2]
```diff
@@ -1,37 +1,72 @@
...
+  issueAnswer: Joi.string().required(),
+  storyAnswer: Joi.string().required(),
+  mentorDescribeAnswer: Joi.string().required(),
+
+  // about you 2 (required for basic profile)
   photoID: Joi.object({
     fileName: Joi.string().required(),
     isPrivate: Joi.boolean().default(true),
   }).required(),
+  hearAboutPressPadAnswer: Joi.string().required(),
```
Comment (MohammedYehia) [inline server/middlewares/validation/internCompleteProfileSchema.js]: should validate for length as you said on the front end page ` no more than 50 words`

File: server/middlewares/validation/internCompleteProfileSchema.js [hunk 3]
```diff
@@ -1,37 +1,72 @@
...
+    email: Joi.required(),
+    phoneNumber: Joi.string().required(),
+  }),
+  DBSCheck: Joi.object({
+    fileName: Joi.string().required(),
+    isPrivate: Joi.boolean().default(true),
+  }),
 
+  // demographic (required for booking)
+  sexualOrientation: Joi.string().required(),
```
Comment (MohammedYehia) [inline server/middlewares/validation/internCompleteProfileSchema.js]: also, I guess you should use the same validation as gender

**Comment by RamyAlshurafa** [comment]
@MohammedYehia 
thanks for your review, but this validation schema just check for the already stored data in the database, and this doesn't handle the error's messages to the front.

**Pull request merged**